flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSourceStatisticsReportTest.java

```diff
@@ -0,0 +1,393 @@
+    @Test
+    public void testFlinkParquetFormatHiveTableSourceStatisticsReport() throws Exception {
+        tEnv.executeSql(
+                String.format(
+                        "create table hive.db1.parquetTable ( %s ) stored as parquet",
+                        String.join(", ", ddlTypesMapToStringList(ddlTypesMap()))));
+        tEnv.executeSql(
+                        String.format(
+                                "insert into hive.db1.parquetTable select * from %s.%s.%s",
+                                catalogName, dbName, sourceTable))
+                .await();
+
+        // Hive to read Parquet file
+        FlinkStatistic statistic =
+                getStatisticsFromOptimizedPlan(
+                        "select * from hive.db1.parquetTable where f_smallint > 100");
+        assertHiveTableParquetFormatTableStatsEquals(statistic.getTableStats(), 3, 1L);
+    }
+
+    @Test
+    public void testMapRedOrcFormatHiveTableSourceStatisticsReport() throws Exception {
+        // Use mapRed parquet format.
+        tEnv.getConfig().set(HiveOptions.TABLE_EXEC_HIVE_FALLBACK_MAPRED_READER, true);
+        tEnv.executeSql(
+                String.format(
+                        "create table hive.db1.orcTable ( %s ) stored as orc",
+                        String.join(", ", ddlTypesMapToStringList(ddlTypesMap()))));
+        tEnv.executeSql(
+                        String.format(
+                                "insert into hive.db1.orcTable select * from %s.%s.%s",
+                                catalogName, dbName, sourceTable))
+                .await();
+
+        // Hive to read Orc file
+        FlinkStatistic statistic =
+                getStatisticsFromOptimizedPlan(
+                        "select * from hive.db1.orcTable where f_smallint > 100");
+        assertHiveTableOrcFormatTableStatsEquals(statistic.getTableStats(), 3, 1L);
+    }
+
+    @Test
+    public void testMapRedParquetFormatHiveTableSourceStatisticsReport() throws Exception {
+        // Use mapRed parquet format.
+        tEnv.getConfig().set(HiveOptions.TABLE_EXEC_HIVE_FALLBACK_MAPRED_READER, true);
+        tEnv.executeSql(
+                String.format(
+                        "create table hive.db1.parquetTable ( %s ) stored as parquet",
+                        String.join(", ", ddlTypesMapToStringList(ddlTypesMap()))));
+        tEnv.executeSql(
+                        String.format(
+                                "insert into hive.db1.parquetTable select * from %s.%s.%s",
+                                catalogName, dbName, sourceTable))
+                .await();
+
+        // Hive to read Parquet file.
+        FlinkStatistic statistic =
+                getStatisticsFromOptimizedPlan(
+                        "select * from hive.db1.parquetTable where f_smallint > 100");
+        assertHiveTableParquetFormatTableStatsEquals(statistic.getTableStats(), 3, 1L);
+    }
+
+    @Test
+    public void testHiveTableSourceWithLimitPushDown() {
```

Comment:
add a case with limit push down and statistics is unknown

Reply:
> add a case with limit push down and statistics is unknown

done!